var/spack/repos/builtin/packages/metall/package.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+class Metall(CMakePackage):
+    """An allocator for persistent memory"""
+
+    homepage = "https://github.com/LLNL/metall"
+    git      = "https://github.com/LLNL/metall.git"
+
+    maintainers = ['KIwabuchi', 'rogerpearce', 'mayagokhale']
+
+    version('develop', branch='develop')
+
+    # Install only header files (no binary files)
+    depends_on('boost@1.64: atomic=False chrono=False clanglibcpp=False context=False coroutine=False date_time=False debug=False exception=False fiber=False filesystem=False graph=False icu=False iostreams=False locale=False log=False math=False mpi=False multithreaded=False numpy=False pic=False program_options=False python=False random=False regex=False serialization=False shared=False signals=False singlethreaded=False system=False taggedlayout=False test=False thread=False timer=False versionedlayout=False wave=False', type=('build', 'link'))
```

Comment:
You should use `~atomic` instead of `atomic=False`. Also, is it not possible to use a version of boost compiled with the default variants enabled? Then users won't have to install multiple copies of boost.

Reply:
Yes, we can also use a version of boost compiled with the default variants.
As Metall needs only header files in Boost, I turned off all binaries to speed up the installation.
What do you think about adding a new option such as '+boost_header_only'? If the option is not given, Spack installs Boost with the default variants.

Reply:
A user can specify that on the command-line (or in a config file) by using:
```console
$ spack install metall ^boost~foo~bar...
```
Unfortunately there's no shortcut at the moment. `depends_on` statements in Spack are hard requirements, not suggestions. As is, you're saying that even if someone already has `boost` installed, they can't use it.

Reply:
Just for clarification, I was thinking of something like this:
`
$ spack install metall # In addition to Metall, build and install Boost with the default variants
$ spack install metall +boost_header_only # In addition to Metall, install only header files in Boost
`
Anyway, I changed to install Boost with the default variants. Please let me know if you have any suggestions.
Best,